goalc/compiler/compilation/Static.cpp

```diff
@@ -557,16 +557,26 @@ Val* Compiler::compile_bitfield_definition(const goos::Object& form,
  */
 StaticResult Compiler::compile_static_no_eval_for_pairs(const goos::Object& form,
                                                         Env* env,
-                                                        int seg) {
+                                                        int seg,
+                                                        bool can_macro) {
   auto fie = env->file_env();
   if (form.is_pair()) {
     if (form.as_pair()->car.is_symbol() && (form.as_pair()->car.as_symbol()->name == "new" ||
                                             form.as_pair()->car.as_symbol()->name == "the" ||
                                             form.as_pair()->car.as_symbol()->name == "lambda")) {
       return compile_static(form, env);
     }
-    auto car = compile_static_no_eval_for_pairs(form.as_pair()->car, env, seg);
-    auto cdr = compile_static_no_eval_for_pairs(form.as_pair()->cdr, env, seg);
+    // try as a macro
+    if (can_macro && form.as_pair()->car.is_symbol()) {
+      goos::Object macro_obj;
+      if (try_getting_macro_from_goos(form.as_pair()->car, &macro_obj)) {
```

Comment:
I think this is fine, we just have to watch out that no macro names conflict with things we actually want in static pairs.

Reply:
you're right, `send-event` comes to mind, I don't think there's any examples of a pair starting with `send-event` thankfully. I think one change we could make later is have it so `'(send-event 1 2 3)` is a pair but `',(send-event ...)` or `'(foo bar ,(send-event ...))` is recognized as a macro. We'll see.